src/Dataset.ts

```diff
@@ -32,20 +33,17 @@ export abstract class Dataset<T extends Tile> {
 
   get extent() : Rectangle {
     return {
-      x: [-1e16, 1e16],
-      y: [-1e16, 1e16],
+      x: [-Infinity, Infinity],
+      y: [-Infinity, Infinity],
     }
   }
 
-  download_most_needed_tiles(bbox : Rectangle, max_ix: number, queue_length : number) {
-    throw new Error('Tile download not implemented');
-  }
-
-  map(callback : (tile: T) => any, after = false) {
+  map<U extends any>(callback : (tile: T) => U, after = false) : U[] {
     // perform a function on each tile and return the values in order.
```

Comment:
Comments that describe the behavior of a method should be JSDoc comments placed above the method